test/elements/proxy.test.ts

```diff
@@ -0,0 +1,180 @@
+import { Chance } from 'chance';
+import { delegateProxy } from '@/elements/services/proxy';
+import { BasisTheoryElementsInternal } from '@/types/elements';
+import { Proxy as ElementsProxy } from '@/types/elements/services';
+import { Proxy, ProxyRequestOptions } from '@/types/sdk';
+
+describe('elements proxy service', () => {
+  let chance: Chance.Chance;
+  let expectedResponse: string;
+  const hasElement = jest.fn();
+  let serviceInstance: Proxy & ElementsProxy;
+  let expectedOptions: ProxyRequestOptions;
+  const superMethod = jest.fn();
+  let elementsInstance: {
+    proxy: {
+      get: jest.Mock;
+      post: jest.Mock;
+      put: jest.Mock;
+      patch: jest.Mock;
+      delete: jest.Mock;
+    };
+  } & {
+    hasElement: jest.Mock;
+  };
+
+  describe('elements is initialized', () => {
+    beforeEach(() => {
+      chance = new Chance();
+
+      expectedResponse = chance.string();
+
+      expectedOptions = {
+        apiKey: chance.string(),
```

Comment:
my mind keeps wondering if it's weird to be able to have the SDK set the right header for an API key but not for proxy key or url. probably not a big deal though

Reply:
lol yeah I thought the same, will wait for user feedback for this one